src/aws-manager.js

```diff
@@ -267,7 +256,15 @@ class AwsManager {
         // Now let's classify them
         for (let reservation of response[0].data.Reservations) {
           for (let instance of reservation.Instances) {
-            let workerType = this.parseKeyPairName(instance.KeyName).workerType;
+            // We want to be able to transition from old (key name) to the new
+            // system based on spot request ids
+            if (!instance.SpotInstanceRequestId) {
+              continue;
+            }
+            let workerType = await this.workerTypeForSRID(instance.SpotInstanceRequestId);
```

Comment:
change `workerTypeForSRID` to be `determineWorkerType(instance)`... then  `determineWorkerType(instance)` will find the workerType based on `spotRequestId` or by looking up instance id...

And `determineWorkerType(instance)` would return `null`, if it's not managed by the aws-provisioner...